core/src/main/java/org/axonframework/commandhandling/AnnotationCommandHandlerAdapter.java

```diff
@@ -58,12 +60,29 @@ public AnnotationCommandHandlerAdapter(Object annotatedCommandHandler) {
      * @param annotatedCommandHandler  The object containing the @CommandHandler annotated methods
      * @param parameterResolverFactory The strategy for resolving handler method parameter values
      */
-    @SuppressWarnings("unchecked")
     public AnnotationCommandHandlerAdapter(Object annotatedCommandHandler,
                                            ParameterResolverFactory parameterResolverFactory) {
+        this(annotatedCommandHandler,
+             parameterResolverFactory,
+             ClasspathHandlerDefinition.forClass(annotatedCommandHandler.getClass()));
```

Comment:
I assume the constructors which use the `ClasspathHandlerDefinition` and reference the new constructor you've introduced which have a `HandlerDefinition`, are kept in place for backwards compatibility, right?

First had the idea that this ticket would move all that class loading stuff to Axon's config. That would thus mean these calls wouldn't be made in the `AnnotationCommandHandlerAdapter`/`DisruptorCommandBus`/`AnnotatedAggregateMetaModelFactory`/etc. But, like I said, I assume you didn't do that for backwards compatibility. Correct?

Reply:
Yes, you are correct!

Reply:
Woohoo, I understood it! ;p